Added a Nix flake containing everything needed to build, dev and run …

MichaelFisher1997 opened the pull request:
This is a Nix Flake for NixOS and anyone with nix package manager installed on their system.

It allow people to enter into the flake dev shell as followed:
`nix develop` and it will install all the listed dependencies to build and run redot avoiding the pain of having to workout what dependencies you need to install and what missing.

MichaelFisher1997 commented:
might need a get job thats a container with nix that can do the following

```
nix develop
scons platform=linuxbsd
```

MichaelFisher1997 commented:
closed as theoparis still links to this as a reference but that one should be used going forward

SkogiB commented:
We working with this one now? We are not sending PRs up to Godot by default at this time, due to a reassessment in the core team, so we can start looking at stuff like this soon.

MichaelFisher1997 commented:
I spoke to McDubh about adding this one in for now a few days ago so i reopened it.
A lot of people have been asking for it.
I know this one works for NixOS and Nix on other linux distros, the PR that expands on this, has support for things like Mac hardware and Arm so that can be added in a little later down the road.
(I do actually have a couple people With Macs that said they will help and test it)

But yeah theirs no rush on it

phanirithvij commented:
just a suggestion, you can rebase instead of adding all these merge commits.

Arctis-Fireblight commented:
Hi @MichaelFisher1997 ,
I am a relatively new maintainer here, and am going back through all of the old PRs.
I see there are a few PRs here related to Nix packaging. While I am somewhat familiar with what Nix is, this is not my area of expertise.
Would you mind explaining your intended use case with this to better help myself and the rest of the team better understand what the benefits of providing upstreamed package definitions?
We are looking to make a final decision regarding this trio of PRs shortly. ( #954 , #247 , #154 )

Arctis-Fireblight commented:
Hey @MichaelFisher1997,
I'm going over the last of the open PRs.
Can you tell me at a high level the differences between this PR vs #954 ?

I am still a little on the fence about incorporating Nix packaging into the main repo, but we probably need to come to a final decision one way or another here soon, along with which of the two PRs will get merged if either of them.

MichaelFisher1997 commented:
I've updated the branch with some improvements:

1. **Fixed nixpkgs URL** - Changed from FlakeHub to canonical GitHub URL (`github:NixOS/nixpkgs/nixpkgs-unstable`) as suggested by CodeRabbit
2. **Removed duplicate dependencies** - Cleaned up the deps list (mono, wayland-scanner, speechd, dotnet-sdk_8 were listed twice)
3. **Added one-command build/run** - You can now simply run `nix run .` to build and start Redot (or use `nix develop` for manual control)
4. **Updated README** - Added comprehensive Nix build instructions as the recommended method

**Confirmed working**: I've tested this on NixOS with `nix run .` - it builds and runs Redot successfully with all dependencies provided by Nix.

The branch is up to date with master and ready to merge when you're ready to make a decision on the Nix-related PRs.

Thanks!

MichaelFisher1997 commented:
@Arctis-Fireblight Done! Rebased onto `dev` - clean linear history with 8 commits, no more merge commits. Ready for review/merge when CI passes.

MichaelFisher1997 commented:
@coderabbitai Thanks for the feedback! I've addressed both issues in commit 312cbc0656:

**1. Hard-coded architecture fix:**
- Added dynamic architecture detection in `flake.nix` using `pkgs.stdenv.hostPlatform.isAarch64`
- The binary name now correctly resolves to `arm64` on ARM64 systems (Apple Silicon, ARM Linux) and `x86_64` on x86_64 systems

**2. Platform target mismatch fix:**
- Updated README manual instructions to show both platform options (`linuxbsd` for Linux, `macos` for macOS)
- Added a note explaining that `nix run .` automatically detects platform and architecture

The flake now correctly supports all four system types:
- `x86_64-linux` → `redot.linuxbsd.editor.x86_64`
- `aarch64-linux` → `redot.linuxbsd.editor.arm64`
- `x86_64-darwin` → `redot.macos.editor.x86_64`
- `aarch64-darwin` → `redot.macos.editor.arm64`
